File: src-tauri/backend/scripts/shouldWatch.ts
import fs from "fs";

import config from "../config/index.ts";

function shouldWatch(filePath: string) {
    const lower = filePath.toLowerCase();
    if (config.rules.ignore.some(folder => lower.includes(folder))) return false;
    if (fs.existsSync(filePath) && fs.lstatSync(filePath).isDirectory()) return true;
    if (filePath.endsWith(".ini")) return true;
    return false;
}

export default shouldWatch;

File: src/backend/scripts/helpers/should_watch.py
import os

from src.backend.scripts.config import config

def should_watch(file_path: str) -> bool:
    lower = file_path.lower()

    for folder in config["rules"]["ignore"]:
        if folder in lower:
            return False

    if os.path.exists(file_path) and os.path.isdir(file_path):
        return True

    return False